core/templates/pages/classroom-admin-page/classroom-admin-page.component.ts

```diff
@@ -86,12 +93,44 @@ export class ClassroomAdminPageComponent implements OnInit {
   classroomViewerMode: boolean = false;
   classroomEditorMode: boolean = false;
   classroomDataSaveInProgress: boolean = false;
+  classroomDataPublishInProgress: boolean = false;
+  classroomDataUnpublishInProgress: boolean = false;
 
   newTopicCanBeAdded: boolean = false;
   topicWithGivenIdExists: boolean = true;
   topicDependencyEditOptionIsShown: boolean = false;
   editTopicOptionIsShown: boolean = true;
 
+  topicsToClassroomRelation: TopicClassroomRelationDict[] = [];
+  filteredTopicsToClassroomRelation: TopicClassroomRelationDict[] = [];
+  validationErrors: string[] = [];
+
+  thumbnailParameters: ImageUploaderParameters = {
+    disabled: false,
+    maxImageSizeInKB: 100,
+    imageName: 'Thumbnail',
+    orientation: 'portrait',
+    bgColor: 'transparent',
+    allowedBgColors: ['transparent'],
+    allowedImageFormats: ['svg'],
+    aspectRatio: '4:3',
+  };
+  bannerParameters: ImageUploaderParameters = {
+    disabled: false,
+    maxImageSizeInKB: 1024,
+    imageName: 'Banner',
+    orientation: 'landscape',
+    bgColor: 'transparent',
+    allowedBgColors: ['transparent'],
+    allowedImageFormats: ['png', 'jpeg'],
+    aspectRatio: '16:9',
+  };
+
+  addNewTopicToClassroom(topicId: string): void {
+    this.onNewTopicInputModelChange();
```

Comment:
It's a bit weird to call a `onXXXModelChange()` inside another function in the component. It is usually meant to be called directly from the template when an event is detected. Should we just update onNewTopicInputModelChange() such that it "adds topic id" in addition to what it is doing already?

Reply:
I agree. Done.